lib/dpl/provider/surge.rb

```diff
@@ -0,0 +1,34 @@
+module DPL
+  class Provider
+    class Surge < Provider
+      npm_g 'surge'
+      context.shell "git rev-parse && cd \"$(git rev-parse --show-toplevel)\"" # Go to repo root 
```

Comment:
Avoid changing working directories without user's knowledge. There are other tasks that could run after the deployment, and they may assume to be in a certain directory.

Use `pushd` and `popd`.